tools/clang/lib/SPIRV/ExecutionModel.h

```diff
@@ -0,0 +1,62 @@
+//===------- ExecutionModel.h - get/set SPV Execution Model -----*- C++ -*-===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//===----------------------------------------------------------------------===//
+//
+//  This file defines a execution model class that defines a mapping between
+//  hlsl::ShaderModel::Kind and spv::ExecutionModel
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef EXECUTION_MODEL_H
+#define EXECUTION_MODEL_H
+
+#include "dxc/DXIL/DxilShaderModel.h"
+#include "spirv/unified1/GLSL.std.450.h"
+#include "clang/SPIRV/SpirvType.h"
+
+namespace clang {
+namespace spirv {
+
+/// Execution Model class.
+class ExecutionModel {
+public:
+  using ShaderKind = hlsl::ShaderModel::Kind;
+  using ExecModel = spv::ExecutionModel;
+
+  bool IsPS() const { return execModel == ExecModel::Fragment; }
+  bool IsVS() const { return execModel == ExecModel::Vertex; }
+  bool IsGS() const { return execModel == ExecModel::Geometry; }
+  bool IsHS() const { return execModel == ExecModel::TessellationControl; }
+  bool IsDS() const { return execModel == ExecModel::TessellationEvaluation; }
+  bool IsCS() const { return execModel == ExecModel::GLCompute; }
+  bool IsRay() const {
+    return execModel >= ExecModel::RayGenerationNV &&
+           execModel <= ExecModel::CallableNV;
+  }
```

Comment:
We can't use hlsl::ShaderModel because there are no ShaderModel object for RT profiles in DxilShaderModel.cpp (rgen_6_4, miss_6_4, etc.).
So I implemented a similar class (ExecutionModel) which would allow me to track RT shader stages.
Nonetheless, I have removed ExecutionModel class and used ShaderModel::Kind to track current entry point.
Let me know if this change works for you.